### PR TITLE
Add the feature to deployer to support GCS path used in flag --filename and --output

### PR DESCRIPTION
**Pull request by ChrisGe4**
@joonlim @brandonnelson3

### CODE REVIEW

File: gke-deploy/core/resource/resource.go [hunk 1]
```diff
@@ -148,28 +148,28 @@ func ParseConfigs(ctx context.Context, configs string, oss services.OSService, r
 // If any lines in a resource object's string representation contain a key in
 // lineComments, the corresponding value will be added as a comment at the end of
 // the line.
```
Comment (joonlim) [inline gke-deploy/core/resource/resource.go]: Maybe just add a sentence saying what is being returned?

File: gke-deploy/core/resource/resource_test.go [hunk 2]
```diff
@@ -1476,7 +1476,7 @@ default                  DaemonSet                test-app-daemonset
 foobar                   Deployment               test-app                          Yes      
 default                  Namespace                foobar                            Yes      
 test-local-deploy-all    ReplicationController    test-app-replicationcontroller    Yes      
-foobar                   Service                  test-app                          Yes      http://34.74.85.152
+foobar                   Service                  test-app                          Yes      34.74.85.152
```
Comment (joonlim) [inline gke-deploy/core/resource/resource_test.go]: Does the unit test pass?

File: gke-deploy/deployer/deployer.go [hunk 3]
```diff
@@ -26,6 +27,10 @@ const (
 	managedByLabelKey  = "app.kubernetes.io/managed-by"
 
 	managedByLabelValue = "gcp-cloud-build-deploy"
+	//temporary folder pattern used to create tmp folders for the files downloaded from GCS
+	K8sConfigStagingDir = "gke_deployer_temp_"
```
Comment (joonlim) [inline gke-deploy/deployer/deployer.go]: deployer -> deploy

File: gke-deploy/services/clients.go [hunk 4]
```diff
@@ -43,6 +46,11 @@ type RemoteService interface {
 	Image(ref name.Reference) (v1.Image, error)
 }
 
+//GcsService is an interface for interacting with Google Cloud Storage
```
Comment (joonlim) [inline gke-deploy/services/clients.go]: space after //

File: gke-deploy/deployer/deployer.go (hunk 3 above)
Comment (joonlim) [reply gke-deploy/deployer/deployer.go]: Do we maybe want to differentiate between the input staging dir and the output staging dir?

Comment (ChrisGe4) [reply gke-deploy/deployer/deployer.go]: TempDir() creates a directory with a random string at the end of the name.

File: gke-deploy/core/resource/resource_test.go (hunk 2 above)
Comment (ChrisGe4) [reply gke-deploy/core/resource/resource_test.go]: the file with the change is in another changelist. I checked in the file.

File: gke-deploy/core/resource/resource.go (hunk 1 above)
Comment (ChrisGe4) [reply gke-deploy/core/resource/resource.go]: Done.

File: gke-deploy/services/clients.go (hunk 4 above)
Comment (ChrisGe4) [reply gke-deploy/services/clients.go]: Done.

File: gke-deploy/deployer/deployer.go [hunk 5]
```diff
@@ -34,12 +39,31 @@ type Deployer struct {
 	UseGcloud bool
 }
 
+var s *gcs.GCS
```
Comment (joonlim) [inline gke-deploy/deployer/deployer.go]: Is this variable used?

Comment (ChrisGe4) [reply gke-deploy/deployer/deployer.go]: removed

**Comment by joonlim** [comment]
LGTM

**Pull request merged**